### PR TITLE
Override ResponseJson::MarkFailed to set application_data_.

### PR DESCRIPTION
**Pull request by jonsimantov**
Setting application_data_ to an empty flatbuffer will prevent application_data_ from being nullptr. This was causing an issue in #737 when AuthResponse::error_code and AuthResponse::error_message were accessing a null application_data_ if the response failed.

***

Integration tests to be run.
***

### Type of Change
Place an `x` the applicable box:
- [X] Bug fix. Add the issue # below if applicable.
- [ ] New feature. A non-breaking change which adds functionality.
- [ ] Other, such as a build process or documentation change.
***

#### Notes
- Bug fixes and feature changes require an update to the `Release Notes` section of `release_build_files/readme.md`.
- Read the contribution guidelines [CONTRIBUTING.md](https://github.com/firebase/firebase-cpp-sdk/blob/main/CONTRIBUTING.md).
- Changes to the public API require an internal API review. If you'd like to help us make Firebase APIs better, please propose your change in a feature request so that we can discuss it together.

### CODE REVIEW

**Comment by dconeybe** [comment]
FWIW I don't think that this is the right fix. I think the right fix is to override `MarkFailed()` in `ResponseJson`. I considered implementing the same fix (check for application_data_ being empty) in #615 but ended up fixing a lower-level problem in `ResponseJson::MarkCompleted()`.

**Comment by jonsimantov** [comment]
> FWIW I don't think that this is the right fix. I think the right fix is to override `MarkFailed()` in `ResponseJson`. I considered implementing the same fix (check for application_data_ being empty) in #615 but ended up fixing a lower-level problem in `ResponseJson::MarkCompleted()`.

What should MarkFailed put in application_data_, just an empty flatbuffer?

**Review by dconeybe — APPROVED** [review]
LGTM. Just needs a release notes entry.

**Review by dconeybe — APPROVED** [review]

File: release_build_files/readme.md [hunk 1]
```diff
@@ -666,7 +672,7 @@ code.
         ([#429](https://github.com/firebase/firebase-cpp-sdk/pull/429)).
     -   AdMob (iOS): Temporarily pinned AdMob dependency to a special version of the
         Google-Mobile-Ads-SDK Cocoapod, "7.69.0-cppsdk", to maintain compatibility
-	with version 8.x of the Firebase iOS SDK.
+        with version 8.x of the Firebase iOS SDK.
```
Comment (jonsimantov) [inline release_build_files/readme.md:675]: Untabified the file.

**Pull request merged**